app/shop.py

```diff
@@ -0,0 +1,41 @@
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: list[int],
+            products: dict
+    ) -> None:
+        self.name = name
+        self.location = location
+        self.products = products
+        self.each_product_cost = {}
+        self.total_cost = 0
+
+    def purchase(self, product_cart: dict) -> int:
+        for product in product_cart:
+            try:
+                product_cost = product_cart[product] * self.products[product]

```

Comment:
This line assumes that `self.products[product]` contains the price of the product. Ensure that `self.products` is structured such that the values are indeed prices.